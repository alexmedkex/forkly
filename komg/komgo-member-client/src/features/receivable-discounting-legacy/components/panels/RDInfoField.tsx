import React from 'react'
import styled from 'styled-components'
import { Field } from '../../../trades/components/Field'
import { rdDiscountingSchema } from '../../utils/constants'
import { findFieldFromSchema } from '../../../../store/common/selectors/displaySelectors'

export interface IRDInfoFieldProps {
  fieldName: string
  value: React.ReactNode
}

export const RDInfoField: React.FC<IRDInfoFieldProps> = props => {
  const { fieldName, value } = props

  return (
    <StyledField data-test-id={`rdInfo-field-component-${fieldName}`}>
      <StyledLabel>{findFieldFromSchema('title', fieldName, rdDiscountingSchema)}</StyledLabel>
      <StyledValue data-test-id={`rdInfo-field-value-${fieldName}`}>{value}</StyledValue>
    </StyledField>
  )
}

const StyledField = styled(Field)`
  display: block;
  margin-bottom: 10px;
`

export const StyledLabel = styled.p`
  font-weight: bold;
  margin-bottom: 0.4rem;
`

export const StyledValue = styled.p`
  white-space: normal;
  padding-bottom: 0.4rem;
`
