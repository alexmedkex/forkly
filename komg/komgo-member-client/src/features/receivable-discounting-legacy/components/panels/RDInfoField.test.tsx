import React from 'react'
import { RDInfoField, IRDInfoFieldProps } from './RDInfoField'
import { render } from '@testing-library/react'

describe('RDInfoField', () => {
  let testProps: IRDInfoFieldProps
  beforeEach(() => {
    testProps = {
      fieldName: 'invoiceAmount',
      value: 100000
    }
  })

  it('renders correctly', () => {
    expect(render(<RDInfoField {...testProps} />).asFragment()).toMatchSnapshot()
  })
})
