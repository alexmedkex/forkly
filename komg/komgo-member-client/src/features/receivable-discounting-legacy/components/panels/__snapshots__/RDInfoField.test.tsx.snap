// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RDInfoField renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  margin: 0;
  line-height: 2;
  white-space: nowrap;
  margin-bottom: 5px;
  max-width: 720px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 {
  display: block;
  margin-bottom: 10px;
}

.c2 {
  font-weight: bold;
  margin-bottom: 0.4rem;
}

.c3 {
  white-space: normal;
  padding-bottom: 0.4rem;
}

<li
    class="c0 c1"
    data-test-id="rdInfo-field-component-invoiceAmount"
  >
    <p
      class="c2"
    >
      Invoice amount
    </p>
    <p
      class="c3"
      data-test-id="rdInfo-field-value-invoiceAmount"
    >
      100000
    </p>
  </li>
</DocumentFragment>
`;
